diff --git a/dlls/winegstreamer/unixlib.c b/dlls/winegstreamer/unixlib.c
--- dlls/winegstreamer/unixlib.c
+++ dlls/winegstreamer/unixlib.c
@@ -214,23 +214,7 @@
     GstCaps *caps;
     gchar *str;
 
-    if (!(caps = gst_type_find_helper_for_data_with_extension(NULL, data, size,
-            extension ? extension + 1 : NULL, &probability)))
-    {
-        GST_ERROR("Failed to detect caps for url %s, data %p, size %u", url, data, size);
-        return NULL;
-    }
-
-    str = gst_caps_to_string(caps);
-    if (probability > GST_TYPE_FIND_POSSIBLE)
-        GST_INFO("Detected caps %s with probability %u for url %s, data %p, size %u",
-                str, probability, url, data, size);
-    else
-        GST_FIXME("Detected caps %s with probability %u for url %s, data %p, size %u",
-                str, probability, url, data, size);
-    g_free(str);
-
-    return caps;
+    return NULL;
 }
 
 GstPad *create_pad_with_caps(GstPadDirection direction, GstCaps *caps)
