From 5f19f38096754765ab6acb2015ef2a18e40e5a1e Mon Sep 17 00:00:00 2001
From: Claire Girka <claire@sitedethib.com>
Date: Sat, 2 Jul 2022 12:15:47 +0200
Subject: [PATCH 5/9] [draft] winepulse: Store PulseAudio device's sysfs path
 when available.

---
 dlls/winepulse.drv/pulse.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/dlls/winepulse.drv/pulse.c b/dlls/winepulse.drv/pulse.c
index 944cf938bdc..db4289777da 100644
--- a/dlls/winepulse.drv/pulse.c
+++ b/dlls/winepulse.drv/pulse.c
@@ -104,6 +104,7 @@ typedef struct _PhysDevice {
     UINT index;
     REFERENCE_TIME min_period, def_period;
     WAVEFORMATEXTENSIBLE fmt;
+    char *sysfs_path;
     char pulse_name[0];
 } PhysDevice;
 
@@ -169,6 +170,8 @@ static void free_phys_device_lists(void)
     do {
         LIST_FOR_EACH_ENTRY_SAFE(dev, dev_next, *list, PhysDevice, entry) {
             free(dev->name);
+            if (dev->sysfs_path)
+                free(dev->sysfs_path);
             free(dev);
         }
     } while (*(++list));
@@ -541,6 +544,7 @@ static void fill_device_info(PhysDevice *dev, pa_proplist *p)
     dev->bus_type = phys_device_bus_invalid;
     dev->vendor_id = 0;
     dev->product_id = 0;
+    dev->sysfs_path = NULL;
 
     if (!p)
         return;
@@ -557,6 +561,9 @@ static void fill_device_info(PhysDevice *dev, pa_proplist *p)
 
     if ((buffer = pa_proplist_gets(p, PA_PROP_DEVICE_PRODUCT_ID)))
         dev->product_id = strtol(buffer, NULL, 16);
+
+    if ((buffer = pa_proplist_gets(p, "sysfs.path")))
+        dev->sysfs_path = strdup(buffer);
 }
 
 static void pulse_add_device(struct list *list, pa_proplist *proplist, int index, EndpointFormFactor form,
-- 
2.42.2

